File: App/lib/models/sale.dart
enum OrderStatus { placed, accepted, shipped, delivered }

class Sale {
  const Sale({
    required this.id,
    required this.date,
    required this.totalAmount,
    required this.status,
    required this.itemsCount,
  });

  final String id;
  final DateTime date;
  final double totalAmount;
  final OrderStatus status;
  final int itemsCount;

  String get statusLabel {
    switch (status) {
      case OrderStatus.placed:
        return 'Order Placed';
      case OrderStatus.accepted:
        return 'Accepted by Distributor';
      case OrderStatus.shipped:
        return 'Shipped by Company';
      case OrderStatus.delivered:
        return 'Delivered';
    }
  }
}

File: Backend/src/main/java/com/agripoultry/entity/Sale.java
package com.agripoultry.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "sales")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Sale {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "sale_id")
    private Integer saleId;

    // The farmer who placed the order
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "farmer_id", nullable = false)
    private User farmer;

    // The distributor who accepted the order
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "distributor_id", nullable = false)
    private User distributor;

    @Column(name = "total_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal totalAmount;

    @Column(name = "paid_amount", precision = 10, scale = 2)
    private BigDecimal paidAmount = BigDecimal.ZERO;

    @Column(name = "due_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal dueAmount;

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    private SaleStatus status = SaleStatus.UNPAID;

    @Column(name = "sale_date")
    private LocalDateTime saleDate;

    @OneToMany(mappedBy = "sale", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<SaleItem> saleItems;

    public enum SaleStatus {
        PAID, PARTIAL, UNPAID
    }
}